### DISCUSSION TITLE
Out of calendar manufacturing order

### DISCUSSION THREAD

**Discussion by zipus** [frePPLe/frepple#582] [Q&A]
Happy new year to anyone reading!
Today I'm here to ask about confirmation about one behavior I detected and not found documented, so I don't know if it's done on purpose a behavior, something to improve, or if I need to activate some flag or if I'm missing something.
I have a MO which was not done in the planned time (planned for today and the worker could not start this task), so I've populated a manufacturing order on approved status, at start=tomorrow 1am. (in order to try to create some replan for tomorrow).
Note my calendar of all my resources is only from 9am to 5pm.
The result after the planning is a confirmed order. When I was expecting some planning as the statement "The manufacturing order is present in the ERP system but can still be rescheduled by frePPLe (optimization input).".
I also checked that if the date for this MO is on calendar-available hours it keeps the MO as it is.
So it's normal that frepple confirms something I said that it's still on approved?
Should I be doing something different to replan a MO?

**Comment by jdetaeye** [comment]
> So it's normal that frepple confirms something I said that it's still on approved?

No, normally it's not right that a MO switches status from approved to confirmed.
Only case I can think of where it makes sense is when the parent MO in a routing is confirmed. This will automatically also bring its step MO in that status as well.

> Should I be doing something different to replan a MO?

Note that the replanning capabilities are far more advanced on the Enterprise Edition.
The Community Edition only has the portions of the planning algorithm to *propose/create new* MO/PO/DO. It doesn't have the logic to *move existing/approved* MO/PO/DO around.
Bottom line: your mileage may vary...

An example to illustrate: In the Enterprise Edition, an approved MO that wasn't completed on time will automatically be moved forward to a new date in the future.   This move is then also propagated to other approved MO that need the material produced by the first MO, and the complete chain of MOs is moved to new dates where they are material and capacity feasible.
In the Community Edition you can handle the first level propagation (by updating the start or end date as you did), but the multi-level propagation will not be working.

Another example: When supplier delays his promised delivery of a PO, the Enterprise Edition will move the complete chain of approved MO that depend on the supplier's delivery.   The Community Edition doesn't do that propagation either.

**Reply by zipus** [reply]
> No, normally it's not right that a MO switches status from approved to confirmed. Only case I can think of where it makes sense is when the parent MO in a routing is confirmed. This will automatically also bring its step MO in that status as well.

Not this case, I'll keep investigating it.

**Reply by zipus** [reply]
> > So it's normal that frepple confirms something I said that it's still on approved?
> 
> No, normally it's not right that a MO switches status from approved to confirmed. Only case I can think of where it makes sense is when the parent MO in a routing is confirmed. This will automatically also bring its step MO in that status as well.

I think I found the problem. I have a set routing operation A. with children operations (1, 2, 3, 4). But in MO I have created:
A aproved
A.1 completed
A.2 completed
A.3 completed
A.4 aproved
**A.2 aproved (as step 2 has errors and need to be done again)**

So last A2. was switched by frepple to confirmed.

Does this behavior make sense for you? How should I allow and plan also de extra operation due to production errors?

**Reply by zipus** [reply]
Something similar happens when I have the following configuration, but in this case the step is not part of the routing, so similar example:
I have a set routing operation A. with children operations **(1, 2, 3)**. But in MO I have created:
A aproved
A.1 completed
A.2 completed
A.3 completed
**A.4 aproved**

So A4. is no defined as operation for route operation A. But it's populated as a MO, because in only this case is necesary a 4th step.
In this case frepple removes the MO, A4.
How should the extra operation which is not in the route be managed to be planned?

**Reply by jdetaeye** [reply]
> **A.2 aproved (as step 2 has errors and need to be done again)**
> So last A2. was switched by frepple to confirmed.
> Does this behavior make sense for you? How should I allow and plan also de extra operation due to production errors?

I haven't reproduced the case, but indeed that doesn't right to me.

> I have a set routing operation A. with children operations **(1, 2, 3)**. But in MO I have created: A aproved A.1 completed A.2 completed A.3 completed **A.4 aproved**
> 
> So A4. is no defined as operation for route operation A. But it's populated as a MO, because in only this case is necesary a 4th step. In this case frepple removes the MO, A4. How should the extra operation which is not in the route be managed to be planned?

Haven't reproduced the case, but frepple should accept your MO that deviates from the official routing.
